File: src/p006/p006.hs
module P006 where

sumOfSquares :: Integer -> Integer
sumOfSquares n = n * (n + 1) * (2 * n + 1) `div` 6

squareOfSum :: Integer -> Integer
squareOfSum n = arithmeticSum ^ 2
  where
    arithmeticSum = n * (n + 1) `div` 2

seriesSolution :: Integer -> Integer
seriesSolution n = squareOfSum n - sumOfSquares n

main :: IO ()
main = do
  print $ seriesSolution 100

File: src/p006/p006.scala
def sumOfSquares(n: Int): Int =
  n * (n + 1) * (2 * n + 1) / 6

def squareOfSum(n: Int): Int =
  val arithmeticSum = n * (n + 1) / 2
  arithmeticSum * arithmeticSum

def seriesSolution(n: Int): Int =
  squareOfSum(n) - sumOfSquares(n)

@main def main(): Unit =
  println(seriesSolution(100))
